search-provider/eks-article-metadata-query-provider.c

```diff
@@ -0,0 +1,433 @@
+
+  for (GSList *l = (GSList *) models; l; l = l->next)
+    {
+      EkncContentObjectModel *model = l->data;
+
+      g_variant_builder_open (&builder, G_VARIANT_TYPE("a{sv}"));
+
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "title",
+                                                      G_VARIANT_TYPE ("s"));
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "content-type",
+                                                      G_VARIANT_TYPE ("s"));
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "thumbnail-uri",
+                                                      G_VARIANT_TYPE ("s"));
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "ekn-id",
+                                                      G_VARIANT_TYPE ("s"));
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "child-tags",
+                                                      G_VARIANT_TYPE ("as"));
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "tags",
+                                                      G_VARIANT_TYPE ("as"));
+
+      g_variant_builder_close (&builder);
+    }
+
+  eks_article_metadata_complete_query (state->provider->skeleton,
+                                       state->invocation,
+                                       (const gchar * const *) shards_strv,
+                                       g_variant_builder_end (&builder));
+}
+
+static GValue *
+append_construction_prop (const gchar *key,
+                          GType        type,
+                          GArray      *values_array,
+                          GPtrArray   *props_array,
+                          guint       *nprops)
+{
+  guint index = values_array->len;
+  g_array_set_size (values_array, values_array->len + 1);
+  g_value_init (&g_array_index (values_array, GValue, index),
+                type);
+  g_ptr_array_add (props_array, g_strdup (key));
+  ++(*nprops);
+
+  return &g_array_index (values_array, GValue, index);
+}
+
+static EkncQueryObject *
+create_query_from_dbus_query_parameters (GVariant    *query_parameters,
+                                         const gchar *application_id)
+{
+  g_autoptr(EkncQueryObject) query = g_object_new (EKNC_TYPE_QUERY_OBJECT, NULL);
```

Comment:
This is unused

Reply:
Indeed - now that I think of it, I think I started on trying to set the properties directly on the object and then noticed they were construct-only and went the other way instead ;-)